### ISSUE TITLE
test: pmempool_check/TEST20: SETUP (all/pmem/debug/memcheck)

### ISSUE THREAD

**Issue by lukaszstolarczuk** [pmem/issues#964]
Found on Fedora, PMDK: 1.5-rc1, on HW

```
$ ndctl list -Nu
(...)  {
    "dev":"namespace0.0",
    "mode":"devdax",
    "map":"dev",
    "size":"123.04 GiB (132.12 GB)",
    "uuid":"90f79574-a42f-43f2-b835-9e2ea9b47c90",
    "chardev":"dax0.0"
  }
```

```
$ ./RUNTESTS pmempool_check -s TEST20 -t all
pmempool_check/TEST20: SETUP (all/pmem/debug/memcheck)
error: '/dev/dax0.0' -- cannot find a contiguous region of given size
error: creating pool file failed
pmempool_check/TEST20 failed with exit code 1.
memcheck20.log below.
pmempool_check/TEST20 memcheck20.log ==75624== Memcheck, a memory error detector
(...)
pmempool_check/TEST20 memcheck20.log ==75624== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

pmem20.log below.
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:236 out_init] pid 75624: program: /home/jenkins/Luke/pmdk/src/tools/pmempool/pmempool
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:238 out_init] libpmem version 1.1
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:242 out_init] src version: 1.5-rc1
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:250 out_init] compiled with support for Valgrind pmemcheck
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:255 out_init] compiled with support for Valgrind helgrind
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:260 out_init] compiled with support for Valgrind memcheck
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:265 out_init] compiled with support for Valgrind drd
pmempool_check/TEST20 pmem20.log <libpmem>: <1> [out.c:270 out_init] compiled with support for shutdown state
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [mmap.c:66 util_mmap_init]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [libpmem.c:56 libpmem_init]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [pmem.c:719 pmem_init]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:419 pmem_init_funcs]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:368 pmem_cpuinfo_to_funcs]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:372 pmem_cpuinfo_to_funcs] clflush supported
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:281 use_avx_memcpy_memset] avx supported
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:285 use_avx_memcpy_memset] PMEM_AVX not set or not == 1
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [pmem.c:216 pmem_has_auto_flush]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [os_auto_flush_linux.c:106 check_domain_in_region] region_path: /sys/bus/nd/devices/region0
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [os_auto_flush_linux.c:58 check_cpu_cache] domain_path: /sys/bus/nd/devices/region0/persistence_domain
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:472 pmem_init_funcs] Flushing CPU cache
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:487 pmem_init_funcs] using clflush
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [init.c:501 pmem_init_funcs] using movnt SSE2
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [pmem_posix.c:107 pmem_os_init]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [libpmem.c:69 libpmem_fini]
pmempool_check/TEST20 pmem20.log <libpmem>: <3> [mmap.c:100 util_mmap_fini]

Last 30 lines of pmemobj20.log below (whole file has 52 lines).
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:258 util_file_get_size] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:223 util_file_get_type] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:132 util_file_exists] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:68 device_dax_size] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:1737 util_poolset_single] path /dev/dax0.0 filesize 132118478848 create 0
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:223 util_file_get_type] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:132 util_file_exists] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file_posix.c:291 util_file_device_dax_alignment] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file_posix.c:212 device_dax_alignment] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:2030 util_poolset_files_local] set 0x75fd8a0 minpartsize 2097152 create 1
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:1815 util_part_open] part 0x75fd9e0 minsize 2097152 create 1
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:132 util_file_exists] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:551 util_file_open] path "/dev/dax0.0" size 0x1ffeffed20 minsize 2097152 flags 2
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:258 util_file_get_size] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:223 util_file_get_type] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:132 util_file_exists] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [file.c:68 device_dax_size] path "/dev/dax0.0"
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:2642 util_replica_map_local] set 0x75fd8a0 repidx 0 flags 1
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [mmap_posix.c:153 util_map_hint] len 132118478848 req_align 0
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <1> [set.c:2675 util_replica_map_local] cannot find a contiguous region of given size
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:628 util_poolset_close] set 0x75fd8a0 del 1
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:2908 util_replica_close] set 0x75fd8a0 repidx 0
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:482 util_unmap_part] part 0x75fd9e0
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:520 util_poolset_free] set 0x75fd8a0
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <2> [obj.c:1371 pmemobj_createU] cannot create pool or pool set
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [libpmemobj.c:65 libpmemobj_fini]
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [obj.c:342 obj_fini]
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:136 util_remote_fini]
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [set.c:191 util_remote_unload]
pmempool_check/TEST20 pmemobj20.log <libpmemobj>: <3> [mmap.c:100 util_mmap_fini]

pmemlog20.log below.
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:236 out_init] pid 75624: program: /home/jenkins/Luke/pmdk/src/tools/pmempool/pmempool
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:238 out_init] libpmemlog version 1.1
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:242 out_init] src version: 1.5-rc1
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:250 out_init] compiled with support for Valgrind pmemcheck
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:255 out_init] compiled with support for Valgrind helgrind
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:260 out_init] compiled with support for Valgrind memcheck
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:265 out_init] compiled with support for Valgrind drd
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <1> [out.c:270 out_init] compiled with support for shutdown state
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <3> [mmap.c:66 util_mmap_init]
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <3> [libpmemlog.c:118 libpmemlog_init]
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <3> [libpmemlog.c:130 libpmemlog_fini]
pmempool_check/TEST20 pmemlog20.log <libpmemlog>: <3> [mmap.c:100 util_mmap_fini]

pmemblk20.log below.
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:236 out_init] pid 75624: program: /home/jenkins/Luke/pmdk/src/tools/pmempool/pmempool
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:238 out_init] libpmemblk version 1.1
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:242 out_init] src version: 1.5-rc1
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:250 out_init] compiled with support for Valgrind pmemcheck
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:255 out_init] compiled with support for Valgrind helgrind
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:260 out_init] compiled with support for Valgrind memcheck
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:265 out_init] compiled with support for Valgrind drd
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <1> [out.c:270 out_init] compiled with support for shutdown state
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <3> [mmap.c:66 util_mmap_init]
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <3> [libpmemblk.c:118 libpmemblk_init]
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <3> [libpmemblk.c:130 libpmemblk_fini]
pmempool_check/TEST20 pmemblk20.log <libpmemblk>: <3> [mmap.c:100 util_mmap_fini]

pmemcto20.log below.
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:236 out_init] pid 75624: program: /home/jenkins/Luke/pmdk/src/tools/pmempool/pmempool
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:238 out_init] libpmemcto version 1.0
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:242 out_init] src version: 1.5-rc1
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:250 out_init] compiled with support for Valgrind pmemcheck
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:255 out_init] compiled with support for Valgrind helgrind
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:260 out_init] compiled with support for Valgrind memcheck
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:265 out_init] compiled with support for Valgrind drd
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <1> [out.c:270 out_init] compiled with support for shutdown state
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <3> [mmap.c:66 util_mmap_init]
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <3> [libpmemcto.c:58 libpmemcto_init]
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <3> [libpmemcto.c:70 libpmemcto_fini]
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <3> [cto.c:132 cto_fini]
pmempool_check/TEST20 pmemcto20.log <libpmemcto>: <3> [mmap.c:100 util_mmap_fini]

pmempool20.log below.
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:236 out_init] pid 75624: program: /home/jenkins/Luke/pmdk/src/tools/pmempool/pmempool
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:238 out_init] libpmempool version 1.3
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:242 out_init] src version: 1.5-rc1
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:250 out_init] compiled with support for Valgrind pmemcheck
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:255 out_init] compiled with support for Valgrind helgrind
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:260 out_init] compiled with support for Valgrind memcheck
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:265 out_init] compiled with support for Valgrind drd
pmempool_check/TEST20 pmempool20.log <libpmempool>: <1> [out.c:270 out_init] compiled with support for shutdown state
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [mmap.c:66 util_mmap_init]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [libpmempool.c:69 libpmempool_init]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [set.c:121 util_remote_init]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [libpmempool.c:85 libpmempool_fini]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [set.c:191 util_remote_unload]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [set.c:136 util_remote_fini]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [set.c:191 util_remote_unload]
pmempool_check/TEST20 pmempool20.log <libpmempool>: <3> [mmap.c:100 util_mmap_fini]

RUNTESTS: stopping: pmempool_check/TEST20 failed, TEST=all FS=any BUILD=debug
```

**Comment by lukaszstolarczuk** [comment]
#3424

**Comment by lukaszstolarczuk** [comment]
@lukaszstolarczuk please retest this please

**Comment by marcinslusarz** [comment]
This is fixed in 1.6 and I don't plan to backport it to 1.5.x.

**Comment by lukaszstolarczuk** [comment]
verified on 1.5-313-g5cdc05152
